### PR TITLE
Add TSIS-1 HSRS solar irradiance spectra

### PR DESCRIPTION
**Pull request by nollety**
# Description

TSIS-1 HSRS solar irradiance spectra (Version 1, Version 2 and Full Spectrum Extension, at all sampling resolutions) were made available via either the data submodule, or the Eradiate FTP server.

The present pull request:
* updates the data submodule, which makes these solar irradiance spectra available within Eradiate
  * the data submodule update also includes [this commit](https://github.com/eradiate/eradiate-data/commit/f57dfda1e4fa346f749ea184d64faaf74b9f0814) which changes all SRF datasets so that their data variables are stored on single float instead of double float.
* updates the data guide page on solar irradiance spectra:
  * re-organise the already existing information
  * add information about newly added spectra
  * fix minor errors 

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I updated the change log if relevant
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

**Comment by nollety** [comment]
@leroyvn, asking your review in particular so that you are aware that I changed the default solar irradiance spectrum.

**Comment by nollety** [comment]
run Eradiate CI

**Comment by nollety** [comment]
Hi @leroyvn, thanks for your feedback!

> your submodule update does more than just add the irradiance spectra: what is it?

That's true; the submodule update moves the submodule forward by two commits, [the first commit](https://github.com/eradiate/eradiate-data/commit/f57dfda1e4fa346f749ea184d64faaf74b9f0814) being an update of the variable data types of the SRF datasets (from double to single float) and the second commit updates the solar irradiance spectra.

Would you like that I make another pull request to move forward the data submodule by the first commit?

**Comment by leroyvn** [comment]
No, I think you can add one more change log entry and document it in this PR.

File: src/eradiate/scenes/spectra/_solar_irradiance.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ class SolarIrradianceSpectrum(Spectrum):
         "(see :ref:`sec-user_guide-data-solar_irradiance` for the list); ",
         type="Dataset",
         init_type="Dataset or str or path-like, optional",
-        default='"thuillier_2003_extrapolated"',
+        default='"coddington_2022-1_nm"',
```
[the remark this answers is not in the document]
Comment (nollety) [reply src/eradiate/scenes/spectra/_solar_irradiance.py:124]: As far as I understand, only the Version 1 (2021) is recommended by the committee on Earth observation satellites (see [here](https://lasp.colorado.edu/lisird/data/tsis1_hsrs/)). This is a typo, thanks for catching that!

**Pull request merged**